nats_test.go

```diff
@@ -1651,6 +1651,107 @@ func TestRevokedUserCredentials(t *testing.T) {
 	wg.Wait()
 }
 
+func TestExpiredAccountAuthentication(t *testing.T) {
```

Comment:
Think some of these tests should be refactored into table driven tests since too similar, I'll update the PR